enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,480 @@
+	// FromCIDR is the IPv4/IPv6 CIDR from which we apply node firewall rule
+	// +kubebuilder:validation:one-of[]=Type=string;Format:ipv4;Format:ipv6
+	FromCIDR string `json:"fromCIDR" protobuf:"bytes,1,opt,name=fromCIDR"`
+}
+
+// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols
+type IngressNodeFirewallICMPRule struct {
+	// ICMPType define ICMP Type Numbers (RFC 792).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=43
+	// +kubebuilder:validation:Minimum:=0
+	ICMPType int8 `json:"icmpType" protobuf:"varint,1,opt,name=icmpType"`
+
+	// ICMPCode define ICMP Code ID (RFC 792).
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Maximum:=16
+	// +kubebuilder:validation:Minimum:=0
+	ICMPCode int8 `json:"icmpCode" protobuf:"varint,2,opt,name=icmpCode"`
+}
+
+// IngressNodeFirewallProtoRule define ingress node firewall rule for TCP, UDP and SCTP protocols
+type IngressNodeFirewallProtoRule struct {
+    // start-end range of dstPorts or [port1, port2,..., portn].
+    // +kubebuilder:validation:Optional
+    // +optional
+    Ports []int32 `json:"ports" protobuf:"varint,2,opt,name=ports"`
+}
+
+// IngressNodeFirewallProtocolRule define ingress node firewall rule per protocol
+type IngressNodeFirewallProtocolRule struct {
+    // IngressNodeFirewallProtoRule define ingress node firewall rule for TCP, UDP and SCTP protocols.
+	ProtocolRule IngressNodeFirewallProtoRule json:"protoRule" protobuf:"varint,1,opt,name=protoRule"`
+    
+	// IngressNodeFirewallICMPRule define ingress node firewall rule for ICMP and ICMPv6 protocols.
+    ICMPRule IngressNodeFirewallICMPRule
+
+	// Protocol can be ICMP, ICMPv6, TCP, SCTP or UDP.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="ICMP";"ICMPv6";"TCP";"UDP";"SCTP"
+	Protocol IngressNodeFirewallRuleProtocolType
+
+	// Action can be Allow or Deny.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum="Allow";"Deny"
+	Action IngressNodeFirewallActionType `json:"action" protobuf:"bytes,4,opt,name=action,casttype=IngressNodeFirewallActionType"`
+}
+
+// ProtocolType defines the protocol types that are supported
+type IngressNodeFirewallRuleProtocolType string
+
+const (
+	// ProtocolTypeICMP refers to the ICMP protocol
+	ProtocolTypeICMP ProtocolType = "ICMP"
+
+	// ProtocolTypeICMPv6 refers to the ICMPv6 protocol
+	ProtocolTypeICMPv6 ProtocolType = "ICMPv6"
+
+	// ProtocolTypeTCP refers to the TCP protocol, for either IPv4 or IPv6
+	ProtocolTypeTCP ProtocolType = "TCP"
+
+	// ProtocolTypeUDP refers to the UDP protocol, for either IPv4 or IPv6
+	ProtocolTypeUDP ProtocolType = "UDP"
+
+	// ProtocolTypeSCTP refers to the SCTP protocol, for either IPv4 or IPv6
+	ProtocolTypeSCTP ProtocolType = "SCTP"
+)
+
+// IngressNodeFirewallActionType indicates whether an IngressNodeFirewallRule allows or denies traffic
+// +kubebuilder:validation:Pattern=`^Allow|Deny$`
```

Comment:
```suggestion
// +kubebuilder:validation:Pattern=`^(Allow|Deny)$`
```
Or simpler:
```suggestion
// +kubebuilder:validation:Enum="Allow";"Deny"
```
But this is superfluous anyway because you already specify validation on the API field that uses this type.